pkg/flow/internal/controller/node_declare_component.go

```diff
@@ -0,0 +1,361 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/module"
+	"github.com/grafana/agent/pkg/flow/logging/level"
+	"github.com/grafana/agent/pkg/flow/tracing"
+	"github.com/grafana/river/ast"
+	"github.com/grafana/river/vm"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/atomic"
+)
+
+// DeclareComponentNode is a controller node which manages a module.
```

Comment:
> manages a module

I think our terminology here (across the set of maintainers, not just you and me) is misaligned, and we should spend a little bit of time to fix that. 

Here's my current view of things:

* Module: a unit of configuration.
* Root module: the configuration directly loaded by `grafana-agent run`. 
* Builtin component: a component provided by the Grafana Agent binary.
* Declare: a configuration block which declares a new component. 
* Declared component: a component provided by a declare block.
* Import: a configuration block which imports component definitions from another module. It is not valid to import a module which contains blocks other than import and declare. 

This is just what I'm suggesting; I think we need to have a offline conversation to get aligned on all the terminology, and then make sure all the comments and type names are aligned with that.

Reply:
the code has been adapted to the terminology decided by the team